sdk/identity/azure-identity/azure/identity/_internal/msal_credentials.py

```diff
@@ -86,3 +86,15 @@ def get_token(self, *scopes):
             raise ClientAuthenticationError(message="authentication failed: {}".format(result.get("error_description")))
 
         return AccessToken(result["access_token"], now + int(result["expires_in"]))
+
+
+class PublicClientCredential(MsalCredential):
+    """Wraps an MSAL PublicClientApplication with the TokenCredential API"""
+
+    def __init__(self, **kwargs):
+        # type: (Any) -> None
+        super(PublicClientCredential, self).__init__(
+            app_class=msal.PublicClientApplication,
```

Comment:
Why are we passing around the class of the current instance? Can't we use [`self.__class__`](https://docs.python.org/3/library/stdtypes.html#instance.__class__)?

Reply:
It's a wrapped MSAL class, passed here so `MsalCredential` knows what to instantiate when the user-facing credential needs it. (MSAL splits auth flow implementations into two classes which are constructed identically.)